src/DssProxyRegistry.sol

```diff
@@ -31,23 +32,23 @@ contract DssProxyRegistry {
     }
 
     function build(address owner_) external returns (address payable proxy) {
-        require(proxies[owner_] == address(0), "DssProxyRegistry/proxy-already-registered-to-owner"); // Not allow new proxy if the user already has one and remains being the owner
+        proxy = payable(proxies[owner_]);
+        require(proxy == address(0) || DssProxy(payable(proxy)).owner() != owner_, "DssProxyRegistry/proxy-registered-to-owner"); // Not allow new proxy if the user already has one and remains being the owner
```

Comment:
I have the same orphaning concern here.  After a user who transferred their proxy builds a new one the old proxy will not be tracked in the registry until someone calls claim for it.  Maybe we're fine with that, but it will mean the `proxies` mapping is not authoritative.

It would cost extra, but we could do:

```solidity
if (DssProxy(payable(proxy)).owner() != owner_) {
    claim(proxy);
}
```

Reply:
I feel it is not an issue, we have a mapping which indicates if proxy comes from this registry (`isProxy`). On the other hand we have a mapping which indicates the actual registered proxies (`proxies`).
A proxy could actually remain as orphan indefinitely if it is transferred to someone that already has a proxy. That situation can change if that person transfers their actual proxy and claims the new one or if they transfer back this proxy to someone that doesn't have one.